libs/ha/ha_common_libs_k8s.py

```diff
@@ -1907,3 +1929,45 @@ def form_endpoint_port(pod_obj, pod_list, port_name="rgw-https"):
             pod_ep_dict[pod] = ip_port
 
         return pod_ep_dict
+
+    def create_iam_user_with_header(self, i_d, header):
+        """
+        Function create IAM user with give header info.
+        :param i_d: Int count number for IAM user name creation
+        :param header: Existing header to use for IAM user creation post request
+        :return: None if IAM user REST req fails or Dict response for IAM user successful creation
+        """
+        user = None
+        payload = {}
+        name = f"ha_iam_{i_d}_{time.perf_counter_ns()}"
+        payload.update({"uid": name})
+        payload.update({"display_name": name})
+        LOGGER.info("Creating IAM user request....")
+        endpoint = CSM_REST_CFG["s3_iam_user_endpoint"]
+        resp = self.restapi.rest_call("post", endpoint=endpoint, json_dict=payload,
+                                      headers=header)
+        LOGGER.info("IAM user request successfully sent...")
+        if resp.status_code == HTTPStatus.CREATED:
+            resp = resp.json()
+            user = dict()
+            user.update({resp["keys"][0]["user"]: {
+                "user_name": resp["keys"][0]["user"],
+                "password": S3_CFG["CliConfig"]["s3_account"]["password"],
+                "accesskey": resp["keys"][0]["access_key"],
+                "secretkey": resp["keys"][0]["secret_key"]}})
+        return user
+
+    def delete_iam_user_with_header(self, user, header):
+        """
+        Function delete IAM user with give header info.
+        :param user: IAM user name to be deleted
+        :param header: Existing header to use for IAM user delete request
+        :return: Tuple
+        """
```

Comment:
Same as above, lets make it generic, instead of just one user, lets get dict of users.

Reply:
> Same as above, lets make it generic, instead of just one user, lets get dict of users.

Same as above

Reply:
Please make changes as discussed